nbconvert/filters/markdown.py

```diff
@@ -56,20 +59,33 @@ def markdown2latex(source, markup='markdown', extra_args=None):
     out : string
       Output as returned by pandoc.
     """
-    return pandoc(source, markup, 'latex', extra_args=extra_args)
+    warnings.warn("""`markdown2latex` is deprecated in favor of 
+                  `nbconvert.filters.pandoc.generic_pandoc` with 
+                  appropriate arguments since nbconvert 5.0""")
```

Comment:
Nitpicking: `with appropriate arguments` you can give the appropriate new syntax. Also triple quoted string keep indentation:

```
In [4]: warnings.warn("""`markdown2latex` is deprecated in favor of
   ...:                   `nbconvert.filters.pandoc.generic_pandoc` with
   ...:                   appropriate arguments since nbconvert 5.0""")
/Users/bussonniermatthias/anaconda/bin/ipython:3: UserWarning: `markdown2latex` is deprecated in favor of
                  `nbconvert.filters.pandoc.generic_pandoc` with
                  appropriate arguments since nbconvert 5.0
```

You might want to use textwrap.dedent or auto joining of simple strings.